### DISCUSSION TITLE
How to set "maximum font size"

### DISCUSSION THREAD

**Discussion by machineonamission** [libvips/libvips#3171] [Q&A]
I'm generating some text, i specified `width` and `height`, and I want to give libvips/pango/whatever a *maximum* font size. I want it to draw at the given font size (in pixels) if it fits inside the `width`x`height` box, or downsize it if it won't fit. The current behavior does no resizing when an absolute font size (in pixels) is specified. if the font size is specified in points, it will scale _up_ the text to fill the box, which I don't want.

**Comment by jcupitt** [comment]
Hi @HexCodeFFF,

I think you have to limit the font size by the size of the text box you fit to. 

You can get the font size it selects with the `autofit-dpi` parameter, so maybe reduce height until you get a dpi under your threshold?

**Comment by machineonamission** [comment]
using the `autofit-dpi` parameter was the key: I generate the text once and check if the DPI is below or equal to a threshold, if it's not I regenerate the text
```py
text_prerender, autofit_dict = pyvips.Image.text(
    text,
    font=f"Twemoji Color Emoji,Tahoma Bold 56",
    rgba=True,
    fontfile="rendering/fonts/TAHOMABD.TTF",
    align=pyvips.Align.CENTRE,
    width=w,
    height=h,
    autofit_dpi=True
)
autofit_dpi = autofit_dict["autofit_dpi"]
if autofit_dpi <= 72:
    text = text_prerender
else:
    text = pyvips.Image.text(
        f"text",
        font=f"Twemoji Color Emoji,Tahoma Bold 56",
        rgba=True,
        fontfile="rendering/fonts/TAHOMABD.TTF",
        align=pyvips.Align.CENTRE,
        width=w,
        height=h,
        dpi=72
    )
```
probably not maximally efficient 'cause the first text() call can be discarded but it works

**Reply by jcupitt** [reply]
Oh nice, I was imagining some kind of iterative thing, but of course you can just cap it.
